osbs/tekton.py

```diff
@@ -0,0 +1,709 @@
+            )
+            r = self.os._get(url)
+            check_response(r)
+            return r.content
+
+    def stream_logs(self, container):
+        kwargs = {'follow': True}
+        if container:
+            kwargs['container'] = container
+
+        # If connection is closed within this many seconds, give up:
+        min_idle_timeout = 60
+
+        # Stream logs, but be careful of the connection closing
+        # due to idle timeout. In that case, try again until the
+        # call returns more quickly than a reasonable timeout
+        # would be set to.
+        while True:
+            connected = time.time()
+            url = self.os._build_url(
+                self.api_path,
+                self.api_version,
+                f"pods/{self.pod_name}/log/",
+                **kwargs
+            )
+            try:
+                logger.debug('Streaming logs for container %s', container)
+                response = self.os._get(url, stream=True,
+                                     headers={'Connection': 'close'})
+                check_response(response)
+
+                for line in response.iter_lines():
+                    connected = time.time()
+                    yield line
+            # NOTE1: If self._get causes ChunkedEncodingError, ConnectionError,
+            # or IncompleteRead to be raised, they'll be wrapped in
+            # OsbsNetworkException or OsbsException
+            # NOTE2: If iter_lines causes ChunkedEncodingError
+            # or IncompleteRead to be raised, it'll simply be silenced.
+            # NOTE3: An exception may be raised from
+            # check_response(). In this case, exception will be
+            # wrapped in OsbsException or OsbsNetworkException,
+            # inspect cause to detect ConnectionError.
+            except OsbsException as exc:
+                if not isinstance(exc.cause, requests.ConnectionError):
+                    raise
+            except requests.exceptions.ConnectionError:
+                pass
+
+            idle = time.time() - connected
+            logger.debug("connection closed after %ds", idle)
+            if idle < min_idle_timeout:
+                # Finish output
+                return
+
+            since = int(idle - 1)
+            logger.debug("fetching logs starting from %ds ago", since)
+            kwargs['sinceSeconds'] = since
+
+    def wait_for_pod(self):
+        logger.info("Waiting for pod to start '%s'", self.pod_name)
+        for changetype, pod in self.os.watch_resource(self.api_path, self.api_version, resource_type='pods', resource_name=self.pod_name):
+            try:
+                status = pod['status']['phase']
+            except KeyError:
+                logger.error("Pod does not have any status")
+                continue
+            if status in ['Running', 'Succeeded', 'Failed']:
+                return pod
+            else:
+                logger.debug("Pod is not in the state I am waiting for")
```

Comment:
Would be nice to mention what is the status

Reply:
updated